### ISSUE TITLE
`In` and `NotIn` inconsistent

### ISSUE THREAD

**Issue by andremedeiros** [jellydator/validation#19]
While using the `In` and `NotIn` rules, I've noticed that their implementations are inconsistent. The former uses DeepEqual to verify equality, whereas the latter uses the equality operator. In cases where pointers are being passed with the same underlying data, the former would work, but the latter falls short.

Can we have a discussion on whether `NotIn` should incorporate `DeepEqual`?

**Comment by swithek** [comment]
That's quite strange. These functions haven't been changed since they were implemented/updated in the original repository, so I cannot answer why these inconsistencies exist, but we can certainly discuss whether they should be updated here.

**Comment by swithek** [comment]
Yes, that sounds good to me. Feel free to open a PR whenever you want!
